include/ck/tensor_operation/gpu/device/impl/device_contraction_utils.hpp

```diff
@@ -0,0 +1,73 @@
+// SPDX-License-Identifier: MIT
+// Copyright (c) 2023, Advanced Micro Devices, Inc. All rights reserved.
+
+#pragma once
+
+#include <cassert>
+#include <vector>
+
+#include "ck/ck.hpp"
+
+namespace ck {
+namespace tensor_operation {
+namespace device {
+
+/**
+ * Calculates the maximum number of subsequent elements of the fast changing dimension
+ * that are consecutive in memory.
+ *
+ * Example:
+ *   NumDimM = 2, NumDimK = 3
+ *   A shape =   [  2,   3,  4, 5, 6]
+ *   A strides = [360, 120, 30, 6, 1]
+ *                |  M   |  |   K  |
+ *   It follows from strides that K is FCD and all the subsequent elements of K are consecutive
+ *   in memory.
+ *   But if strides were [360, 120, 6, 24, 1], then only 6 subsequent elements of K would be
```

Comment:
Hmm that was really my intention (not to be confusing but to have swapped dimensions) 😄
In this example when you iterate over memory, your 6 first elements are `[0, 0, 0, 0, 0] - [0, 0, 0, 0, 5]` and then the 7th is `[0, 0, 1, 0, 0]` (instead of `[0, 0, 0, 1, 0]` like it would be in the original case).
So from the other side -- when you iterate over the tensor's elements, first 6 elements are contiguous in memory but the 7th one ([0, 0, 0, 1, 0]) is on the 120th element in memory.

We can discuss it offline.